backend/src/main/java/com/bakdata/conquery/sql/execution/ResultSetProcessor.java

```diff
@@ -0,0 +1,57 @@
+package com.bakdata.conquery.sql.execution;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.bakdata.conquery.models.types.ResultType;
+
+public interface ResultSetProcessor {
```

Comment:
Wenn gewünscht zieh das ich mit in die ResultType Klasse rein über
```java
public abstract T getFromResultSet(ResultSet resultSet, int columnIndex, ResultSetProcessor resultSetProcessor) throws SQLException

// example usage
@CPSType(id = "BOOLEAN", base = ResultType.class)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public static class BooleanT extends PrimitiveResultType<Boolean> {
	// ...		
	@Override
	public Boolean getFromResultSet(ResultSet resultSet, int columnIndex, ResultSetProcessor resultSetProcessor) {
		return resultSetProcessor.getBoolean(resultSet, columnIndex);
	}
}
```

So wie jetzt kann der `ResultType` "ungenerisch" bleiben und ich muss nicht bei allen Usages <?> o.Ä. hinzufügen.

Reply:
Mir wäre es lieber es in den ResultType zu machen, ihr habt bisher aber alles versucht an der Seite implementiert, deswegen ist so auch ok.